ConfigSpace/hyperparameters.pyx

```diff
@@ -1538,88 +1555,129 @@ cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
-            for n in range(self.lower, self.upper + 1):
-                if n != int_value:
-                    if transform:
-                        neighbors.append(n)
-                    else:
-                        n = self._inverse_transform(n)
-                        neighbors.append(n)
+        """Get the neighbors of a value
+
+        NOTE
+        ----
+        **This assumes the value is in the unit-hypercube [0, 1]**
+
+        Parameters
+        ----------
+        value: float
+            The value to get neighbors around. This assume the ``value`` has been
+            converted to the [0, 1] range which can be done with ``_inverse_transform``.
+
+        rs: RandomState
+            The random state to use
+
+        number: int = 4
+            How many neighbors to get
+
+        transform: bool = False
+            Whether to transform this value from the unit cube, back to the
+            hyperparameter's specified range of values.
+
+        std: float = 0.2
+            The std. dev. to use in the [0, 1] hypercube space while sampling
+            for neighbors.
+
+        Returns
+        -------
+        List[int]
+            Some ``number`` of neighbors centered around ``value``.
+        """
+        assert 0 <= value <= 1, (
+            "For get neighbors of UniformIntegerHyperparameter, the value"
+            " if assumed to be in the unit-hypercube [0, 1]. If this was not"
+            " the behaviour assumed, please raise a ticket on github."
+        )
+        # Convert python values to cython ones
+        cdef long long center = self._transform(value)
+        cdef long long lower = self.lower
+        cdef long long upper = self.upper
+        cdef unsigned int n_requested = number
+        cdef unsigned long long n_neighbors = upper - lower - 1
+        cdef long long stepsize = self.q if self.q is not None else 1
+
+        neighbors = []
+
+        cdef long long v  # A value that's possible to return
+        if n_neighbors < n_requested:
+
+            for v in range(lower, center):
+                neighbors.append(v)
+
+            for v in range(center + 1, upper):
+                neighbors.append(v)
+
+            if transform:
+                return neighbors
+            else:
+                return self._inverse_transform(np.asarray(neighbors)).tolist()
+
+
+        # A truncated normal between 0 and 1, centered on the value with a scale of std.
+        # This will be sampled from and converted to the corresponding int value
+        # OPTIM: Use of clipping insead of scipy.truncnorm
+        # Clipping is faster for generation but more likely to generate clumped values of 0/1
+        # at the boundaries, given a wide enough std and a value close to 0 or 1.
+        # However the overhead of the scipy call seems to outweight this clumping + deduping
+        # procedure in our used benchmark.
+        float_indices = np.clip(rs.normal(value, std, n_requested), 0, 1)
```

Comment:
This was actually one of the main culprits for the increased time taken.

Scipy's truncnorm is a good bit slower (at least on the time scale we are interested in). It was faster to just clip the normal, as was done before. This may however be detrimental in the case of a value close to the 0-1 boundary and a large enough stddev.

My guess is with a different benchmark, we would end up with a lot of duplicate 0's (clipped values) which means we have to enter the slower loop part below.